os/net/ipv6/sicslowpan.c

```diff
@@ -1181,13 +1187,18 @@ uncompress_hdr_iphc(uint8_t *buf, uint16_t buf_size, uint16_t ip_len)
       }
     }
     /* if tmp == 0 we do not have a context and therefore no prefix */
-    uncompress_addr(&SICSLOWPAN_IP_BUF(buf)->srcipaddr,
-                    tmp != 0 ? context->prefix : NULL, unc_ctxconf[tmp],
-                    (uip_lladdr_t *)packetbuf_addr(PACKETBUF_ADDR_SENDER));
+    if(uncompress_addr(&SICSLOWPAN_IP_BUF(buf)->srcipaddr,
```

Comment:
I think ìf(!thing)` is the expectation for boolean checks. (Same for the other if-statements.)